test/TwigEnvironmentFactoryTest.php

```diff
@@ -19,59 +19,55 @@
-use Prophecy\Prophecy\ProphecyInterface;
 use Psr\Container\ContainerInterface;
+use ReflectionException;
+use ReflectionProperty;
 use Twig\Environment as TwigEnvironment;
+use Twig\Error\LoaderError;
+use Twig\Error\RuntimeError;
 use Twig\Extension\CoreExtension;
 use Twig\Extension\EscaperExtension;
 use Twig\Extension\OptimizerExtension;
 use Twig\RuntimeLoader\RuntimeLoaderInterface;
 
-use function is_string;
-
 class TwigEnvironmentFactoryTest extends TestCase
 {
-    /**
-     * @var ContainerInterface|ProphecyInterface
-     */
+    /** @var MockObject<ContainerInterface> */
     private $container;
 
-    protected function setUp() : void
+    protected function setUp(): void
     {
-        $this->container = $this->prophesize(ContainerInterface::class);
+        $this->container = $this->createMock(ContainerInterface::class);
     }
 
-    public function testCallingFactoryWithNoConfigReturnsTwigEnvironmentInstance()
+    /**
+     * @throws LoaderError
+     */
+    public function testCallingFactoryWithNoConfigReturnsTwigEnvironmentInstance(): TwigEnvironment
     {
-        $this->container->has('config')->willReturn(false);
-        $this->container->has(TwigExtension::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Twig\TwigExtension::class)->willReturn(false);
-        $this->container->has(ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(UrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\UrlHelper::class)->willReturn(false);
+        $this->container->expects(self::atLeastOnce())->method('has')->withAnyParameters()->willReturn(false);
         $factory     = new TwigEnvironmentFactory();
-        $environment = $factory($this->container->reveal());
+        $environment = $factory($this->container);
 
         $this->assertInstanceOf(TwigEnvironment::class, $environment);
 
         return $environment;
     }
 
-    public function testUsesDebugConfiguration()
+    /**
+     * @throws LoaderError
+     */
+    public function testUsesDebugConfiguration(): void
     {
         $config = ['debug' => true];
-        $this->container->has('config')->willReturn(true);
-        $this->container->get('config')->willReturn($config);
-        $this->container->has(TwigExtension::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Twig\TwigExtension::class)->willReturn(false);
-        $this->container->has(ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(UrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\UrlHelper::class)->willReturn(false);
+        $this->container->expects(self::atLeastOnce())->method('has')->withAnyParameters()->willReturnMap(
```

Comment:
`withAnyParameters()` is default I believe, so it can be dropped here and in other similar uses
```suggestion
        $this->container->expects(self::atLeastOnce())->method('has')->willReturnMap(
```

Reply:
done for all